thinc/layers/softmax.py

```diff
@@ -37,26 +37,38 @@ def Softmax_v2(
     init_W: Callable = zero_init,
     init_b: Callable = zero_init,
     normalize_outputs: bool = True,
+    temperature=1.0,
```

Comment:
Hmm, actually I'm not sure whether the `temperature` is better here in the `Softmax` layer itself or whether it should descend into the `softmax` ops themselves. So I'm not sure where people use `softmax` with temperature outside of the `Softmax` layer, but for example in an `attention` layer we can make the `softmax` sharper for more `hard` decisions. Also maybe we will implement the straight-through gumbel-softmax at some point and then that would not necessarily run the `Softmax` layer, maybe just the `softmax` op itself right?

Reply:
I think this would lead to us handling temperature half in the op (for forward) and half in the op (for backward), since the backward is tailored to the cross-entropy loss case and doesn't use the `backprop_softmax` op (which is for softmax as an activation).

Doing it in the layer would mean that we could always push handling down to the op in the future if we want to, since it wouldn't change the API. However, if we put it in the op now, we have to commit to that API.

Reply:
Doing it in the op would allow more efficient implementations though (getting the maximum value and applying temperature could be fused into one loop).

I still propose to do it as-is now and move it down to the op when needed.

Reply:
Added a commit that applies temperature in the ops. We can always drop the commit before merging the PR if we don't like it.